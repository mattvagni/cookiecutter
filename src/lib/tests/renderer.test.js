const {renderFiles} = require('../renderer');
const {getTemplateConfig} = require('../config');
const fs = require('fs-extra');

jest.mock('../config');

global.console = {
    log: jest.fn(),
    error: jest.fn(),
};

describe('renderFiles()', () => {

    const outputPath = `${__dirname }/test-templates-output`;

    afterEach(() => {
        fs.removeSync(outputPath);
    });

    it('should correctly render the files when the template is just a path', () => {

        getTemplateConfig.mockImplementation(() => {
            return {
                templatePath: `${__dirname }/test-templates/CONTAINER_NAME.js`,
                outputPath,
            };
        });

        renderFiles({
            templateName: '',
            fields: {
                CONTAINER_NAME: 'Baz',
            },
        }, null, process.env.PWD, process.env.PWD);

        expect(fs.readFileSync(`${outputPath }/Baz.js`, 'utf8')).toMatchSnapshot();
    });

    it('should correctly render the files when the template is a folder', () => {

        getTemplateConfig.mockImplementation(() => {
            return {
                templatePath: `${__dirname }/test-templates/COMPONENT_NAME`,
                outputPath,
            };
        });

        renderFiles({
            templateName: '',
            fields: {
                COMPONENT_NAME: 'Bar',
                specialNumber: 'A',
            },
        }, null, process.env.PWD, process.env.PWD);

        expect(fs.readFileSync(`${outputPath }/Bar/index.js`, 'utf8')).toMatchSnapshot();
        expect(fs.readFileSync(`${outputPath }/Bar/styles.css`, 'utf8')).toMatchSnapshot();
        expect(fs.readFileSync(`${outputPath }/Bar/sub-folder/Bar.js`, 'utf8')).toMatchSnapshot();
    });

    it('should throw an error if the output of a template allready exists', () => {

        getTemplateConfig.mockImplementation(() => {
            return {
                templatePath: `${__dirname }/test-templates/COMPONENT_NAME`,
                outputPath,
            };
        });

        function func() {
            renderFiles({
                templateName: '',
                fields: {
                    COMPONENT_NAME: 'Bar',
                    specialNumber: 'A',
                },
            }, null, process.env.PWD, process.env.PWD);
        }

        // Render the template the first time.
        func();

        // Render the template the second time, this time it should error
        // because it already exists.
        expect(func).toThrowErrorMatchingSnapshot();
    });

});
