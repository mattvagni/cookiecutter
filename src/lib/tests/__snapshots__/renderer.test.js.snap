// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderFiles() should correctly render the files when the template is a folder 1`] = `
"import React from 'react';

const number = A;

class Bar React.Component {
    render() {
        return <div>Hello world {number}</div>;
    }
}

export default Bar"
`;

exports[`renderFiles() should correctly render the files when the template is a folder 2`] = `
"/*
Styles for Bar component
*/

.root {
    background: red;
}"
`;

exports[`renderFiles() should correctly render the files when the template is a folder 3`] = `
"console.log('Bar');
"
`;

exports[`renderFiles() should correctly render the files when the template is a folder excluding the containing directory if set 1`] = `
"console.log('Bar');
"
`;

exports[`renderFiles() should correctly render the files when the template is a folder excluding the containing directory if set 2`] = `
"import React from 'react';

const number = A;

class Bar React.Component {
    render() {
        return <div>Hello world {number}</div>;
    }
}

export default Bar"
`;

exports[`renderFiles() should correctly render the files when the template is just a path 1`] = `
"import React from 'react';

class Baz React.Component {
    render() {
        return <div>Im a container</div>;
    }
}

export default Baz"
`;

exports[`renderFiles() should throw an error if the output of a template already exists 1`] = `"src/lib/tests/test-templates-output/Bar/index.js already exists."`;
